code/modules/mob/living/simple_animal/hostile/true_changeling.dm

```diff
@@ -0,0 +1,193 @@
+	var/mob/living/carbon/human/stored_changeling = null //The changeling that transformed
+	var/devouring = FALSE //If the true changeling is currently devouring a human
+	var/wallcrawl = FALSE //If the true changeling is crawling around the place, allowing it to counteract gravity loss
+
+/mob/living/simple_animal/hostile/true_changeling/New()
+	..()
+	if(prob(33))
+		icon_state = "horror2"
+	else if(prob(33))
+		icon_state = "horror3"
+	spawn(0)
+		to_chat(usr, playstyle_string)
+
+/mob/living/simple_animal/hostile/true_changeling/Life()
+	..()
+	adjustBruteLoss(-TRUE_CHANGELING_PASSIVE_HEAL) //True changelings slowly regenerate
+	time_spent_as_true++ //Used for re-forming
+	if(stored_changeling && time_spent_as_true >= TRUE_CHANGELING_FORCED_REFORM)
+		death() //After a while, the ling'll revert back without being able to control it
+
+/mob/living/simple_animal/hostile/true_changeling/Stat()
+	..()
+	if(statpanel("Status"))
+		if(stored_changeling)
+			var/time_left = TRUE_CHANGELING_FORCED_REFORM - time_spent_as_true
+			time_left = Clamp(time_left, 0, INFINITY)
+			stat(null, "Time Remaining: [time_left]")
+		stat(null, "Ignoring Gravity: [wallcrawl ? "YES" : "NO"]")
+
+/mob/living/simple_animal/hostile/true_changeling/death()
+	..(1)
+	if(stored_changeling && mind)
+		visible_message("<span class='warning'>[src] lets out a furious scream as it shrinks into its human form.</span>", \
+						"<span class='userdanger'>We lack the power to maintain this form! We helplessly turn back into a human...</span>")
+		stored_changeling.loc = get_turf(src)
+		mind.transfer_to(stored_changeling)
+		stored_changeling.Unconscious(30) //Make them helpless for 30 seconds
+		stored_changeling.status_flags &= ~GODMODE
+		qdel(src)
+	else
+		visible_message("<span class='warning'>[src] lets out a waning scream as it falls, twitching, to the floor.</span>", \
+						"<span class='userdanger'>We have fallen! We begin the revival process... (this will take around 45 seconds.)</span>")
+		spawn(450)
+			if(src)
+				visible_message("<span class='warning'>[src] suddenly rises from the dead!</span>")
+				revive() //Changelings can self-revive, and true changelings are no exception
+
+/mob/living/simple_animal/hostile/true_changeling/mob_negates_gravity()
+	return wallcrawl
+
+/mob/living/simple_animal/hostile/true_changeling/adjustFireLoss(amount)
+	amount *= 2 //True changelings take 2 times the burn damage!
+	..()
+
+/mob/living/simple_animal/hostile/true_changeling/verb/reform()
+	set name = "Re-Form Human Shell"
+	set desc = "We turn back into a human. This takes considerable effort and will stun us for some time afterwards."
+	set category = "True Changeling"
+
+	if(stat)
+		to_chat(usr, "<span class='warning'>We must be conscious to debase ourselves!</span>")
+		return 0
+	if(!stored_changeling)
+		to_chat(usr, "<span class='warning'>We do not have a form other than this!</span>")
+		return 0
+	if(stored_changeling.stat == DEAD)
```

Comment:
is it really worth having this in here since they're going to get forced back in eventually anyway?

Reply:
true